### PR TITLE
Added Descriptive Aria-Labels to Branch/Tag/Commit Selector

### PR DESCRIPTION
**Pull request by lmelende20**
Regarding issue #35477 - [Accessibility]: Branch/Tag/Commit: 'Find...' input placeholders aren't descriptive.

I added descriptive aria-labels, as instructed in issue #35477, that corresponds to the tab that the user has selected. I tested the changes using the VoiceOver screen reader and the labels are read as desired however, the placeholder text is also being read. Hiding the placeholder text was not explicitly stated in the issue however, if there are some suggestions as to how to accomplish this, it would be appreciated.

## App preview:

- [Web](https://sg-web-lm-branchestagscommitslabels.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-yjeibzsavx.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/web/src/repo/RevisionsPopover/RevisionsPopover.tsx [hunk 1]
```diff
@@ -129,6 +130,7 @@ export const RevisionsPopover: React.FunctionComponent<React.PropsWithChildren<R
                                 repo={props.repo}
                                 currentCommitID={props.currentCommitID}
                                 onSelect={props.onSelect}
+                                tabLabel={tab.id}
```
Comment (jasongornall) [inline client/web/src/repo/RevisionsPopover/RevisionsPopover.tsx:133]: is an id human readable?

Comment (jasongornall) [reply client/web/src/repo/RevisionsPopover/RevisionsPopover.tsx:133]: Maybe this should be tab.label?

File: client/web/src/repo/RevisionsPopover/RevisionsPopoverTab.tsx [hunk 2]
```diff
@@ -36,6 +38,7 @@ export const RevisionsPopoverTab: React.FunctionComponent<React.PropsWithChildre
             autoFocus={true}
             inputPlaceholder="Find..."
             compact={true}
+            inputAriaLabel={`Find ${inputAriaLabel}`}
```
[the remark this answers is not in the document]
Comment (lmelende20) [reply client/web/src/repo/RevisionsPopover/RevisionsPopoverTab.tsx:41]: Find __ was suggested in the issue which is why I chose that, but I can change it so it is more descriptive

[the remark this answers is not in the document]
Comment (jasongornall) [reply client/web/src/repo/RevisionsPopover/RevisionsPopoverTab.tsx:41]: Ahh I I'll let you make the call my fault for not reading the issue

File: client/web/src/repo/RevisionsPopover/RevisionsPopoverReferences.tsx [hunk 3]
```diff
@@ -114,6 +114,8 @@ interface RevisionsPopoverReferencesProps {
     showSpeculativeResults?: boolean
 
     onSelect?: (node: GitRefFields) => void
+
+    tabLabel?: string
```
Comment (jasongornall) [inline client/web/src/repo/RevisionsPopover/RevisionsPopoverReferences.tsx:118]: if this is optional... can this result in Find undefined?

File: client/web/src/repo/RevisionsPopover/RevisionsPopover.tsx (hunk 1 above)
Comment (lmelende20) [reply client/web/src/repo/RevisionsPopover/RevisionsPopover.tsx:133]: `const TABS: RevisionsPopoverTab[] = [
    { id: 'branches', label: 'Branches', noun: 'branch', pluralNoun: 'branches', type: GitRefType.GIT_BRANCH },
    { id: 'tags', label: 'Tags', noun: 'tag', pluralNoun: 'tags', type: GitRefType.GIT_TAG },
    { id: 'commits', label: 'Commits', noun: 'commit', pluralNoun: 'commits' },
]`

tab.id and tab.label are essentially the same. tab.id is all lowercase while tab.label starts with an uppercase. Do you think there are advantages to using tab.label over tab.id?

Comment (jasongornall) [reply client/web/src/repo/RevisionsPopover/RevisionsPopover.tsx:133]: it looks like the id is meant to key off react, in my ideal world we would have a description 😅

File: client/web/src/repo/RevisionsPopover/RevisionsPopoverReferences.tsx (hunk 3 above)
Comment (lmelende20) [reply client/web/src/repo/RevisionsPopover/RevisionsPopoverReferences.tsx:118]: Without line 118, line 136 gets an error: "Property 'tabLabel' does not exist on type 'PropsWithChildren<PropsWithChildren<RevisionsPopoverReferencesProps>>'."

File: client/web/src/repo/RevisionsPopover/RevisionsPopover.tsx [hunk 4]
```diff
@@ -51,14 +51,35 @@ interface RevisionsPopoverTab {
...
-    { id: 'branches', label: 'Branches', noun: 'branch', pluralNoun: 'branches', type: GitRefType.GIT_BRANCH },
-    { id: 'tags', label: 'Tags', noun: 'tag', pluralNoun: 'tags', type: GitRefType.GIT_TAG },
-    { id: 'commits', label: 'Commits', noun: 'commit', pluralNoun: 'commits' },
+    {
+        id: 'branches',
+        label: 'Branches',
+        noun: 'branch',
+        pluralNoun: 'branches',
+        type: GitRefType.GIT_BRANCH,
+        description: 'Find a revision from the listed branches',
```
Comment (jasongornall) [inline client/web/src/repo/RevisionsPopover/RevisionsPopover.tsx:66]: I like it!

**Review by jasongornall — APPROVED** [review]

**Pull request merged**